stream-chat-android-e2e-test/src/main/kotlin/io/getstream/chat/android/e2e/test/rules/RetryRule.kt

```diff
@@ -44,31 +46,84 @@ public class RetryRule(private val count: Int) : TestRule {
         return object : Statement() {
             @Throws(Throwable::class)
             override fun evaluate() {
+                val testName = description.displayName
                 val retryAnnotation: Retry? = description.getAnnotation(Retry::class.java)
                 val retryCount = retryAnnotation?.count ?: count
                 val databaseOperations = DatabaseOperations()
                 var caughtThrowable: Throwable? = null
+                lateinit var videoFilePath: String
+                lateinit var recordingThread: Thread
 
                 for (i in 0 until retryCount) {
                     try {
-                        System.err.println("${description.displayName}: run #${i + 1} started.")
+                        System.err.println("$testName: run #${i + 1} started.")
                         device.executeShellCommand("logcat -c")
+                        videoFilePath = "${Environment.getExternalStorageDirectory().absolutePath}/$testName.mp4"
+                        recordingThread = startVideoRecording(videoFilePath)
                         base.evaluate()
+                        stopVideoRecording(videoFilePath, recordingThread)
                         return
                     } catch (t: Throwable) {
-                        System.err.println("${description.displayName}: run #${i + 1} failed.")
-                        databaseOperations.clearDatabases()
+                        System.err.println("$testName: run #${i + 1} failed.")
                         caughtThrowable = t
+                        databaseOperations.clearDatabases()
+                        stopVideoRecording(videoFilePath, recordingThread)
                         device.allureLogcat(name = "logcat_${i + 1}")
                         device.allureScreenshot(name = "screenshot_${i + 1}")
                         device.allureWindowHierarchy(name = "hierarchy_${i + 1}")
+                        device.allureScreenrecord(
+                            name = "record_${i + 1}",
+                            file = File(videoFilePath),
+                        )
+                    } finally {
+                        device.executeShellCommand("rm $videoFilePath")
                     }
                 }
 
                 throw caughtThrowable ?: IllegalStateException()
             }
         }
     }
+
+    private fun startVideoRecording(remoteVideoPath: String): Thread {
```

Comment:
All changes in this file is just copy-pasta from the Video project testing infra.